src/Kmd.Logic.Gateway.Automation/Publish.cs

```diff
@@ -168,7 +169,12 @@ private async Task CreateOrUpdateProducts(Guid subscriptionId, Guid providerId,
                                 this.publishResults.Add(new GatewayAutomationResult() { ResultCode = ResultCode.CustomPolicyCreated, EntityId = created.Id });
                                 break;
                             case ValidationStatus.CanBeUpdated:
-                                // TODO: Update CustomPolicy
+                                foreach (var custompolicy in policiesResults.CustomPolicies)
```

Comment:
1. You have nested foreach loop in another foreach, that accepts the same array....... Why?
2. Remove this for each. Just update using `customPolicyResult` object from line 159.

Reply:
Using customPolicyResult object now.